huaweicloud/services/cfw/resource_huaweicloud_cfw_protection_rule.go

```diff
@@ -627,17 +603,40 @@ func resourceProtectionRuleRead(_ context.Context, d *schema.ResourceData, meta
 	return diag.FromErr(mErr.ErrorOrNil())
 }
 
-func FilterRules(rules []interface{}, id string) (interface{}, error) {
-	if len(rules) != 0 {
-		for _, v := range rules {
-			rule := v.(map[string]interface{})
-			if rule["rule_id"] == id {
-				return v, nil
-			}
-		}
+func GetProtectionRule(client *golangsdk.ServiceClient, id, objectID string) (interface{}, error) {
```

Comment:
should put this func in acl rule and rename it to GetACLRule
acl_rule is a new resource to replace protection_rule